transcom/mymove
# INT B 18889 remove prime ability to update destination address for approved shipment outside correct endpoint

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18889)

## Summary

When a shipment is in an `APPROVED` status, we do not want the Prime to have the freedom to update the destination address of a shipment without the TOO approving the change first. 

Right now there are two endpoints where the destination address can be updated without the respective logic checks that are in the `updateShipmentDestinationAddress` endpoint. This PR does the following:
- when calling the `updateMTOShipment` and `updateMTOShipmentAddress` endpoints, implemented logic to check the shipment status and if the shipment's destination address is provided in the `payload`
- handled error return to inform the user to use the `updateMTOShipmentDestinationAddress` endpoint instead when trying to update the destination address
- updated tests to reflect changes in logic

### How to test

1. Access MM as prime
2. Find a move with an approved shipment
3. Try to edit the destination address for the shipment by clicking the `Edit` and submitting the form
4. You should get back an error

## Screenshots
![Screenshot 2024-03-14 at 2 42 46 PM](https://github.com/transcom/mymove/assets/136510600/e9ecd425-d542-40fb-b2a0-d7cdcbce87e7)
